### ISSUE TITLE
Test a `Mention` tag

### ISSUE THREAD

**Issue by snarfed** [fediverse-devnet/feditest-tests-fediverse#22]
Deliver a `Create` of a `Note` with a `tag` with type `Mention` and `href` pointing to a user's actor id. Check that that user receives a notification of the `Note`. (This is not specified in AP itself.)

**Comment by jernst** [comment]
This isn't really mentioned in any of the Community Reports currently produced by the SocialWG either, right? Maybe another "glue" report needed?

**Comment by snarfed** [comment]
Maybe! This is probably AS2, including audience targeting, not AP per se, but still.

A bit more background: https://github.com/snarfed/bridgy-fed/issues/887

It may be a bit small for a report on its own. We could combine it with other semir-standardized AS2-centric data model questions like quote posts and profile links and maybe groups? I dunno.

**Comment by jernst** [comment]
In https://www.w3.org/TR/activitystreams-vocabulary/#h-microsyntaxes, it says that to receive notifications, the actor should be in the `to` primary audience, while a mere mention in `tag` should not cause a notification. But I think you are saying that the actor should get notification even in the second case?

**Comment by snarfed** [comment]
I don't know that existing projects follow the AS2 spec very closely (if at all) for UX decisions like what specifically generates a notification. Mastodon, for example, needs a `Mention` tag, but doesn't need that user in either `to` or `cc`: https://github.com/snarfed/bridgy-fed/issues/493#issuecomment-1897843483 . `to`/`cc` are more for visibility than notifications: https://docs.joinmastodon.org/spec/activitypub/#Mention

**Comment by jernst** [comment]
If existing projects notify some actors even if the spec doesn't require it -- and what the spec says sounds reasonable to me -- I don't think we need to test going beyond the spec. If you agree, please close.

**Comment by snarfed** [comment]
The main point here is just to test mentions. Looking for a notification was just the common user-facing way I thought of to check that the mention was handled, like in #26 that search was a common user-facing way to check that a hashtag was handled.

I don't care too much _how_ we check mentions and hashtags. I think it'd be useful to test creating them, which was my goal with both this and #26, but I'll defer to you!

**Comment by jernst** [comment]
Fair enough. I'm actually not clear what the mention tag is supposed to do. If it isn't required to trigger a notification, having it or not having it attached makes which difference where? It's not like applications use it to create a searchable "mention index" or something ... although they certainly could.

**Comment by snarfed** [comment]
Beyond notifications, another use is to link to the mentioned user on the local instance instead of their home instance. For example, try clicking on `@littlealex` in https://mastodon.social/@astridpoot/111862989549373384, then open that post in a different Mastodon instance and try again.

**Comment by jernst** [comment]
Is an instance under any obligation to create a mention tag if somebody types @snarfed? If not, I'm not sure what we can test, other than that any such structure, should it be generated, is syntactically valid. And maybe that whatever is in the mention tag is actually mentioned in the text...

**Comment by snarfed** [comment]
Sure. There's probably more that's possible as soon as you're ready for interactive tests, but at long as you're working purely at the AP S2S level, then yes, not much to do here yet.

**Comment by jernst** [comment]
We have something like this working, this issue will probably get broken into several shortly.
